### ISSUE TITLE
Add 'CurveTween' Node

### ISSUE THREAD

**Issue by Error7Studios** [godotengine/godot-proposals#1251]
**Credits:**
100% credit to Bramwell Williams (@bramreth) for following idea, code, and video.
I'm posting this proposal since he hasn't made one for it.

**Describe the project you are working on:**
(Applies to any project which uses a Tween)

**Describe the problem or limitation you are having in your project:**
There is currently no built-in way to set up a Tween using the inspector.
This also means that you do not get a visualization of the Tween's behavior.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
A new type of `Tween` node which uses a `Curve` resource to define the Tween's behavior directly in the inspector.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
![CurveTweenProposal](https://user-images.githubusercontent.com/60025044/88376916-f2f7bd80-cd63-11ea-91d5-9a8ef45e10c3.png)

**Video demonstration/explanation:**
https://youtu.be/HpiWhvQO_1s

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I think this would be used often as the preferred way to work with Tweens, since it is simple, quick, and visual.

**Is there a reason why this should be core and not an add-on in the asset library?:**
Tweens are often used, and this is a very user-friendly way to create complex Tween behavior directly in the editor.

**Comment by dalexeev** [comment]
Perhaps it is better to add these features to the existing `Tween` class rather than create a new one? And is it necessary to be able to animate an arbitrary curve? Wouldn't a Bezier curve be enough? `AnimationPlayer` only uses Bezier curves.

**Comment by Error7Studios** [comment]
You're right, adding this functionality to the `Tween` class is probably better.
However, wouldn't a `Curve` resource be better than setting up a Bezier curve like in `AnimationPlayer`, since the `Curve `resource could be saved into a variable and reused?

**Comment by eon-s** [comment]
I was thinking on this, like some set of optional defaults for Tween node, it can allow to tweak things visually and make custom easings in a simple way.

**Comment by Calinou** [comment]
With https://github.com/godotengine/godot/pull/41794 being merged, this can no longer be a node.

**Comment by KoBeWi** [comment]
All you need is to export a Curve resource and use method tweening:
```
@export var curve: Curve

func _ready():
    create_tween().tween_method(tween_curve, 0, 1, 1)

func tween_curve(t):
    something = curve.interpolate_baked(t)
```

**Comment by KoBeWi** [comment]
I opened an alternate proposal: #7807
It would make using curves easier and allow to make a general-purpose convenient addon.
